packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -311,6 +202,7 @@ export default function BlockList( {
 			<BlockListItem
 				index={ index }
 				isStackedHorizontally={ isStackedHorizontally }
+				key={ clientId }
```

Comment:
While `keyExtractor` will set this for a virtualized list, the simple, mapped list needs `key` explicitly set.